src/gpl/src/mbff.cpp

```diff
@@ -0,0 +1,1058 @@
+#include "mbff.h"
+#include <iostream>
+#include <algorithm>
+#include <vector>
+#include <map>
+#include <set>
+#include <ctime>
+#include <chrono>
+#include <memory>
+#include <random>
+#include <omp.h>
+#include <ortools/linear_solver/linear_solver.h>
+#include <ortools/base/logging.h>
+#include <ortools/sat/cp_model.h>
+#include <ortools/sat/cp_model.pb.h>
+#include <ortools/sat/cp_model_solver.h>
+#include <lemon/list_graph.h>
+#include <lemon/network_simplex.h>
+#include <lemon/maps.h>
+
+/* 
+height and width of a flip-flop
+ratios are used for:
+    (1) finding all slot locations in a tray
+    (2) finding power consumption per slot 
+*/
+
+float WIDTH = 2.448, HEIGHT = 1.2;
+float RATIOS[6] = { 1, 0.875, 0.854, 0.854, 0.844, 0.844 };
+
+namespace gpl {
+
+int MBFF::GetBitCnt(int bit_idx) { return (1 << bit_idx); }
+
+float MBFF::RunLP(std::vector<Flop> flops, std::vector<Tray> &trays,
+                  std::vector<std::pair<int, int> > clusters, int sz) {
```

Comment:
flops & clusters should be passed by const-ref.  Many other similar cases elsewhere.
sz is unused